tests/config/test_config.py

```diff
@@ -17,13 +32,12 @@ def test_config_get_processes_depended_on_values(config, config_cache_dir):
 @pytest.mark.parametrize(
     ("name", "env_value", "value"),
     [
-        ("installer.parallel", "true", True),
-        ("installer.parallel", "false", False),
-        ("virtualenvs.create", "true", True),
-        ("virtualenvs.create", "false", False),
+        (name, env_value, value)
+        for name in get_boolean_options()
+        for env_value, value in [("true", True), ("false", False)]
     ],
 )
 def test_config_get_from_environment_variable(config, environ, name, env_value, value):
-    env_var = "POETRY_{}".format("_".join(k.upper() for k in name.split(".")))
+    env_var = "POETRY_{}".format(re.sub("[.-]+", "_", name).upper())
```

Comment:
I would move this into the generation of the tupes as this would mean we can see what we are testing in the test case name.